src/test/java/de/tum/in/www1/artemis/exercise/fileuploadexercise/FileUploadExerciseIntegrationTest.java

```diff
@@ -370,6 +367,21 @@ void deleteFileUploadExercise_asInstructor() throws Exception {
         assertThat(exerciseRepo.findByCourseIdWithCategories(course.getId())).isEmpty();
     }
 
+    @Test
+    @WithMockUser(username = TEST_PREFIX + "instructor1", roles = "INSTRUCTOR")
+    void testDeleteFileUploadExerciseWithChannel() throws Exception {
+        Course course = fileUploadExerciseUtilService.addCourseWithFileUploadExercise();
+        FileUploadExercise fileUploadExercise = fileUploadExerciseRepository.findByCourseIdWithCategories(course.getId()).get(0);
+        Channel exerciseChannel = exerciseUtilService.addChannelToExercise(fileUploadExercise);
+
+        request.delete("/api/file-upload-exercises/" + fileUploadExercise.getId(), HttpStatus.OK);
+
+        await().untilAsserted(() -> {
+            Optional<Channel> exerciseChannelAfterDelete = channelRepository.findById(exerciseChannel.getId());
+            assertThat(exerciseChannelAfterDelete).isEmpty();
```

Comment:
are you 100% certain that this will not be a flaky test? This seems fairly similar to the issue we had to fix in the ExamIntegrationTest

Reply:
Has nothing to do with the tests, but I am deleting the channel synchronously now anyway. Before and after these changes I let each new test run at least 2000 times and they never failed.